code/modules/mining/lavaland/necropolis_chests.dm

```diff
@@ -192,6 +194,64 @@
 	icon_state = "asclepius_active"
 	activated = TRUE
 
+//Memento Mori
+/obj/item/clothing/neck/necklace/memento_mori
+	name = "Memento Mori"
+	desc = "A mysterious pendant. An inscription on it says: \"Certain death tomorrow means certain life today.\""
+	icon = 'icons/obj/lavaland/artefacts.dmi'
+	icon_state = "memento_mori"
+	actions_types = list(/datum/action/item_action/hands_free/memento_mori)
+	resistance_flags = INDESTRUCTIBLE | LAVA_PROOF | FIRE_PROOF | UNACIDABLE | ACID_PROOF
+	var/mob/living/carbon/human/active_owner
+
+/obj/item/clothing/neck/necklace/memento_mori/item_action_slot_check(slot)
+	if(slot == SLOT_NECK)
+		return TRUE
+
+/obj/item/clothing/neck/necklace/memento_mori/ui_action_click(mob/user, action)
+	if(!active_owner)
+		if(ishuman(user))
+			memento(user)
+	else
+		to_chat(user, "<span class='warning'>You try to free your lifeforce from the pendant...</span>")
+		if(do_after(user, 40, target = src))
+			mori()
+
+/obj/item/clothing/neck/necklace/memento_mori/dropped(mob/user)
+	..()
+	if(active_owner)
+		mori()
+
+//Just in case
+/obj/item/clothing/neck/necklace/memento_mori/Destroy()
+	if(active_owner)
+		mori()
+	return ..()
+
+/obj/item/clothing/neck/necklace/memento_mori/proc/memento(mob/living/carbon/human/user)
+	to_chat(user, "<span class='warning'>You feel your life being drained by the pendant...</span>")
+	if(do_after(user, 40, target = src))
+		to_chat(user, "<span class='notice'>Your lifeforce is now linked to the pendant! You feel like removing it would kill you, and yet you instinctively know that until then, you won't die.</span>")
+		user.add_trait(TRAIT_NODEATH, "memento_mori")
+		user.add_trait(TRAIT_NOHARDCRIT, "memento_mori")
+		user.add_trait(TRAIT_NOCRITDAMAGE, "memento_mori")
+		icon_state = "memento_mori_active"
+		active_owner = user
+
+/obj/item/clothing/neck/necklace/memento_mori/proc/mori()
+	icon_state = "memento_mori"
+	if(!active_owner)
+		return
+	var/mob/living/carbon/human/H = active_owner //to avoid infinite looping when dust unequips the pendant
+	active_owner = null
+	to_chat(H, "<span class='userdanger'>You feel your life rapidly slipping away from you!</span>")
+	H.dust(TRUE, TRUE)
+
+/datum/action/item_action/hands_free/memento_mori
```

Comment:
I want to open some discussion on the effect triggering with an action button, instead of the moment you put it on. My view is that it's better to call `memento()` when you put it on, to allow people to get stuck with it if they don't know what it is, and also so you can put it on someone, possibly to save them.

Reply:
I'm not a huge fan of noobtrap mechanics in SS13; it creates an unnecessary discrepancy between those who read PRs and those who don't. As for saving people, the action button is usable while unconscious, so it's still a possibility.

Reply:
Just add a yes/no pop up warning like with BoH bombs, ie "Are you sure you want to do this?"

Give a warning of some kind that appropriately conveys what the item's intended use is.

Reply:
An examine revealing what it does would be ok without being invasive like a popup.

Reply:
absolutely horrible idea for a popup, talk about immersion breaking.